### PR TITLE
feat: [CDS-46254]:step_group_template

### PR DESCRIPTION
**Pull request by litvinovstas**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger titlecheck

**Comment by litvinovstas** [comment]
retrigger titlecheck

**Comment by litvinovstas** [comment]
retrigger cypress-cv

File: src/modules/70-pipeline/hooks/useAddStepTemplate.ts [hunk 1]
```diff
@@ -103,7 +108,9 @@ export function useAddStepTemplate(props: AddStepTemplate): AddStepTemplateRetur
         gitDetails,
         storeMetadata
       })
-      const newStepData = { step: createStepNodeFromTemplate(template, isCopied) }
+      const stepType = template.templateEntityType === 'StepGroup' ? 'stepGroup' : 'step'
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/hooks/useAddStepTemplate.ts:111]: Can we use TemplateType enum here ? 
template.templateEntityType === TemplateType.StepGroup ? 

And please check this at other places as well

File: src/modules/70-pipeline/hooks/useAddStepTemplate.ts [hunk 2]
```diff
@@ -126,17 +133,18 @@ export function useAddStepTemplate(props: AddStepTemplate): AddStepTemplateRetur
...
         drawerData: {
           type: DrawerTypes.StepConfig,
           data: {
             stepConfig: {
-              node: newStepData.step,
+              node: newStepData[stepType],
               stepsMap: event.stepsMap,
               onUpdate: executionRef?.stepGroupUpdated,
-              isStepGroup: false,
+              isStepGroup: template?.templateEntityType === 'StepGroup',
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/hooks/useAddStepTemplate.ts:147]: https://github.com/harness/harness-core-ui/pull/13357/files#r1065444848

File: src/modules/70-pipeline/components/PipelineStudio/PipelineContext/PipelineContext.tsx [hunk 3]
```diff
@@ -1010,7 +1010,8 @@ export enum PipelineContextType {
   Pipeline = 'Pipeline',
   StageTemplate = 'StageTemplate',
   PipelineTemplate = 'PipelineTemplate',
-  Standalone = 'Standalone'
+  Standalone = 'Standalone',
+  StepGroupTemplate = 'StegGroupTempalte'
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineContext/PipelineContext.tsx:1014]: StegGroupTempalte -> StepGroupTemplate

spelling error

File: src/modules/72-templates-library/components/TemplateSelector/TemplateSelectorLeftView/TemplateSelectorLeftView.tsx [hunk 4]
```diff
@@ -98,9 +99,11 @@ export const TemplateSelectorLeftView: React.FC<TemplateSelectorLeftViewProps> =
   }, [selectedScope, orgIdentifier])
 
   const body = React.useMemo(() => {
+    const isStepType = templateType === 'Step'
```
Comment (samarth-sharma31) [inline src/modules/72-templates-library/components/TemplateSelector/TemplateSelectorLeftView/TemplateSelectorLeftView.tsx:102]: can we use template type enum here as well -> enum TemplateType.Step

File: src/modules/72-templates-library/components/TemplateSelector/TemplateSelectorLeftView/TemplateSelectorLeftView.tsx [hunk 5]
```diff
@@ -98,9 +99,11 @@ export const TemplateSelectorLeftView: React.FC<TemplateSelectorLeftViewProps> =
...
+    const templateEntityTypes = isStepType ? [templateType, 'StepGroup'] : [templateType]
```
Comment (samarth-sharma31) [inline src/modules/72-templates-library/components/TemplateSelector/TemplateSelectorLeftView/TemplateSelectorLeftView.tsx:103]: Lets connect for a quick call regarding this

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger cypress-rest

**Comment by litvinovstas** [comment]
retrigger jest

**Review by samarth-sharma31 — APPROVED** [review]

**Comment by litvinovstas** [comment]
retrigger jest

**Comment by litvinovstas** [comment]
retrigger cypress-cd

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**